src/Model/batchprocessing/BatchprocessMachineLearningDataSelection.py

```diff
@@ -0,0 +1,135 @@
+from src.Model.batchprocessing.BatchProcess import BatchProcess
+import logging
+import pandas as pd
+import os
+import re
+
+
+class BatchprocessMachineLearningDataSelection(BatchProcess):
+    """
+    This class handles batch processing for the machine learning
+    data selection process. Inherits from the BatchProcessing class.
+    """
+
+    def __init__(self, progress_callback, interrupt_flag,
+                 dvh_data_path, pyrad_data_path, dvh_value, pyrad_value):
+        """
+        Class initialiser function.
+        :param progress_callback: A signal that receives the current
+                                  progress of the loading.
+        :param interrupt_flag: A threading.Event() object that tells the
+                               function to stop loading.
+        :param dvh_data_path: dvh path to file.
+        :param pyrad_data_path: pyradiomics path to file.
+        :param dvh_value: selected value for DVH
+        :param pyrad_value: selected value for Pyradiomics
+        """
+        # Call the parent class
+        super(BatchprocessMachineLearningDataSelection, self).__init__(
+            progress_callback,
+            interrupt_flag,
+            None)
+        self.dvh_data_path = dvh_data_path
+        self.pyrad_data_path = pyrad_data_path
+
+        self.dvh_value = dvh_value
+        self.pyrad_value = pyrad_value
+
+        self.dvh_data = None
+        self.pyrad_data = None
+
+        self.full_path_dvh = None
+        self.full_path_pyrad = None
+
+    def start(self):
```

Comment:
Are you ever checking for the interrupt flag? This is what is called interrupting the processing.

Reply:
Thanks, I removed it